test/module/irohad/consensus/yac/yac_gate_test.cpp

```diff
@@ -120,14 +129,22 @@ TEST_F(YacGateTest, YacGateSubscriptionTest) {
 
   // make blocks
   EXPECT_CALL(*block_creator, on_block())
-      .WillOnce(Return(rxcpp::observable<>::just(expected_block)));
+      .WillOnce(Return(
+          rxcpp::observable<>::just<shared_model::interface::BlockVariant>(
+              expected_block)));
 
   init();
 
   // verify that yac gate emit expected block
   auto gate_wrapper = make_test_subscriber<CallExact>(gate->on_commit(), 1);
-  gate_wrapper.subscribe(
-      [this](auto block) { ASSERT_EQ(*block, *expected_block); });
+  gate_wrapper.subscribe([this](const auto &block_variant) {
+    ASSERT_NO_THROW({
```

Comment:
Maybe it is better to make it as a separate function since it is used 3 times?

Reply:
If it is a function is would be hard to define where assert is thrown. In general it is not recommended to use assert outside gtest